### PR TITLE
RNC-2456 Add KMI deployment step to RAC orb

### PR DESCRIPTION
**Pull request by enicholson125**
This commits the new version tag to the rac-kmi-gitops repo, kicking off a deployment in KMI.

### CODE REVIEW

**Review by jensraaby — APPROVED** [review]
I'm not an expert on CircleCI but this looks sensible to me. 
Maybe @rmorais would have more observations

File: rac-gcp-deploy/orb.yml [hunk 1]
```diff
@@ -765,6 +765,57 @@ jobs:
...
+        default: CIRCLE_PROJECT_REPONAME
+      kmi-env:
+        description: "The environment of KMI to deploy to"
+        type: string
+        default: uat
+      git-username:
+        description: "Name of the environment variable storing the github user to use when pushing commits"
+        type: env_var_name
+        default: GIT_USERNAME
+      git_user_email:
```
Comment (rmorais) [inline rac-gcp-deploy/orb.yml:786]: this uses `_` instead of `-` like the other parameters

**Review by rmorais — APPROVED** [review]
Looks good to me. Just a small nitpick regarding naming :D

File: rac-gcp-deploy/orb.yml [hunk 2]
```diff
@@ -765,6 +765,57 @@ jobs:
...
+      - run:
+          name: Clone gitops repo
+          command: |
+            rm -rf ~/gitops
+            git clone https://$GITHUB_TOKEN@github.com/ovotech/rac-kmi-gitops ~/gitops
+      - run:
+          name: Patch deployment
+          command: |
+            cd ~/gitops/$<<parameters.container-name>>
+            yq e ".version=\"$(cat <<parameters.workspace-dir>>/version.txt)\"" -i values-<<parameters.kmi-env>>-shared.yaml
```
[the remark this answers is not in the document]
Comment (enicholson125) [reply rac-gcp-deploy/orb.yml:804]: It took me a while to track that one down as well! Turns out it's part of the sbt build process - e.g. https://github.com/ovotech/rac-load-profile-coefficients/blob/ab71fa43a463c13d8b4b46c1e6d8aafe25c4c24b/build.sbt#L27 for rac-load-profile-coefficents

File: rac-gcp-deploy/orb.yml (hunk 1 above)
Comment (enicholson125) [reply rac-gcp-deploy/orb.yml:786]: The other jobs in the orb also use `git_user_email` - I can't find anywhere in the ovotech organization that's explicitly setting it though, as it looks like it's only being set through the environment variable, so I've changed it in all the jobs.

File: rac-gcp-deploy/orb.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jensraaby) [reply rac-gcp-deploy/orb.yml:804]: Ah 🤦  I missed that. Good spot!

**Pull request merged**